### ISSUE TITLE
ConcurrentModificationException crash

### ISSUE THREAD

**Issue by dsrees** [dsrees/JavaPhoenixClient#57]
Received this a couple of times. client version 0.2.5
```
Fatal Exception: java.util.ConcurrentModificationException
       at java.util.ArrayList$Itr.next(ArrayList.java:860)
       at org.phoenixframework.Socket.triggerChannelError(Socket.kt:467)
       at org.phoenixframework.Socket.onConnectionError$JavaPhoenixClient(Socket.kt:458)
       at org.phoenixframework.Socket$connect$3.invoke(Socket.kt:229)
       at org.phoenixframework.Socket$connect$3.invoke(Socket.kt:66)
       at org.phoenixframework.WebSocketTransport.onFailure(Transport.kt:134)
       at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:571)
       at okhttp3.internal.ws.RealWebSocket$1.onFailure(RealWebSocket.java:221)
       at okhttp3.RealCall$AsyncCall.execute(RealCall.java:225)
       at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)
```

**Comment by AdamGrzybkowski** [comment]
any news on that? I see this quite often

**Comment by dsrees** [comment]
@AdamGrzybkowski what client version are you on?

**Comment by dsrees** [comment]
fixed in #66, v0.3.1
